src/signer/canister/src/lib.rs

```diff
@@ -128,21 +133,42 @@ async fn sign_prehash(prehash: String) -> String {
 
 /// Returns the Bitcoin address of the caller.
 #[update(guard = "caller_is_not_anonymous")]
-async fn caller_btc_address(network: BitcoinNetwork) -> String {
-    bitcoin_utils::public_key_to_p2pkh_address(
-        network,
-        &bitcoin_utils::ecdsa_pubkey_of(&ic_cdk::caller()).await,
-    )
+async fn caller_btc_address(
+    params: GetAddressRequest,
+) -> Result<GetAddressResponse, GetAddressError> {
+    match params.address_type {
+        BitcoinAddressType::P2WPKH => {
+            let address =
+                bitcoin_utils::principal_to_p2wpkh_address(params.network, &ic_cdk::caller())
+                    .await
+                    .map_err(|msg| GetAddressError::InternalError { msg })?;
+
+            Ok(GetAddressResponse { address })
+        }
+    }
 }
 
 /// Returns the Bitcoin balance of the caller's address.
 #[update(guard = "caller_is_not_anonymous")]
-async fn caller_btc_balance(network: BitcoinNetwork) -> u64 {
-    let address = bitcoin_utils::public_key_to_p2pkh_address(
-        network,
-        &eth::ecdsa_pubkey_of(&ic_cdk::caller()).await,
-    );
-    bitcoin_api::get_balance(network, address).await
+async fn caller_btc_balance(
+    params: GetBalanceRequest,
+) -> Result<GetBalanceResponse, GetBalanceError> {
```

Comment:
Ditto.  If we can make `btc_` as a prefix I think that would make the API easier to use.

Reply:
Discussed - the btc prefix can be added in another PR.